var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -591,6 +598,9 @@ def nvariant_bool(feature):
         if "+advanced_debug" in spec:
             cmake_args.append("-DVTK_DEBUG_LEAKS:BOOL=ON")
 
+        if self.spec.satisfies("@5.11.0-RC2:"):
+            cmake_args.append("-DPARAVIEW_USE_HIP:BOOL=%s" % variant_bool("+hip"))
+
```

Comment:
We need to setup the following vars

``
    Kokkos_CXX_COMPILER:                                                                                                                                                            
    CMAKE_HIP_ARCHITECTURES:
```

Reply:
`Kokkos_CXX_COMPILER` needs to be specified in Kokkos when being transitively used through more than one cmake nested `projects()`. 
CMAKE_HIP_ARCHITECTURES is needed since the compiler default choice sometimes do not execute in target machines like in the ORNL supercomputers. We can delegate this by making this package a subclass of HipPackage (@kwryankrattiger has a mr attempting this)

Reply:
I assume you mean `ROCmPackage` based on other usages. I think I got this right…

Reply:
For some reason we need `Kokkos_CXX_COMPILER` to be set to the hipcc in order to build this.

Reply:
I added a direct dep.

Reply:
I think that it will not resolve the issue, the problem is at cmake level rather than at spack level. You can try to to perform the build of this MR, perhaps it works.

Reply:
Ok, I have a fix based on #33772, but it doesn't do all the fancy logic there. We really need the `hip` package to have an easy query to answer "where is your `clang`?" instead of duplicating everything in the method edited in the linked PR.